From 63e52849ba08a813b30290e14657be6bf019023d Mon Sep 17 00:00:00 2001
From: Min He <min.he@intel.com>
Date: Mon, 14 Mar 2016 20:35:32 +0800
Subject: [PATCH 383/403] Disable PG on SKL as a temp workaround

On latest SKL revision (rev 09), we found that bug 1000(Run mp4 file by WMP
triggers TDR and dom0 panic) happened again. So here we disable power
gating for all the SKL GT3/4 steppings.

This patch is a temp workaround for SKL beta. Will investigate if we can
handle forcewake gracefully in VGT code.

V2: added some comments

Signed-off-by: Min He <min.he@intel.com>
---
 drivers/gpu/drm/i915/intel_pm.c |    7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_pm.c b/drivers/gpu/drm/i915/intel_pm.c
index 9290810..0566f1b 100644
--- a/drivers/gpu/drm/i915/intel_pm.c
+++ b/drivers/gpu/drm/i915/intel_pm.c
@@ -4721,9 +4721,14 @@ static void gen9_enable_rc6(struct drm_device *dev)
 	 * 3b: Enable Coarse Power Gating only when RC6 is enabled.
 	 * WaRsDisableCoarsePowerGating:skl,bxt - Render/Media PG need to be disabled with RC6.
 	 */
+
+	/* Disabling PG on all revisions of SKL GT3/4, since we can see bug 1000
+	 * on latest revision of SKL (rev 09). It's a temp workaround and we
+	 * need to find more graceful method to handle forcewake on VGT.
+	 */
 	if (IS_BXT_REVID(dev, 0, BXT_REVID_A1) ||
 	    ((IS_SKL_GT3(dev) || IS_SKL_GT4(dev)) &&
-	     IS_SKL_REVID(dev, 0, SKL_REVID_F0)))
+	     IS_SKL_REVID(dev, 0, REVID_FOREVER)))
 		I915_WRITE(GEN9_PG_ENABLE, 0);
 	else
 		I915_WRITE(GEN9_PG_ENABLE, (rc6_mask & GEN6_RC_CTL_RC6_ENABLE) ?
-- 
1.7.10.4

